test/DurableTaskEndToEndTests.cs

```diff
@@ -1149,6 +1149,41 @@ private static void CompareTwoComplexTypeObjects(ComplexType firstObject, Comple
             }
         }
 
+        [Fact]
+        public async Task SetStatusOrchestration()
+        {
+            using (JobHost host = TestHelpers.GetJobHost(this.loggerFactory, nameof(this.SetStatusOrchestration)))

```

Comment:
This test is great and it is showing that we can update custom status. Shall we have a simple one covering the basic case on a single set of the status? I think we can reuse `SetStatus` method but also pass the number of iterations. So the simple case will have only a single iteration. I think this is an optional change.

Reply:
Do you think a simple one would cover more than what is already covered here? In general, I prefer to write fewer tests that cover more scenarios to avoid having too many tests.

Reply:
@cgillum Thank you for the clarification. Your are approach is very good. In general I see more value in the integration tests and this is great approach for implementing integration tests.